### PR TITLE
Implements byondStorage for 516 browsers

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Adds support for persistent storage across instances on 516, as added in the past day by lummox.
This won't change anything for 515 clients or lead to data loss until the inevitable move to 516.

## Why It's Good For The Game
Being able to save chat history, settings, etc. is important for 516 players just as much as 515 players.

## Testing
Pre changes: No chat settings/history saved.
Post changes: Chat settings/history saved.

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -257,6 +257,10 @@
 	//CONNECT//
 	///////////
 /client/New(TopicData)
+	// TODO: Remove with 516
+	if(byond_version >= 516) // Enable 516 compat browser storage mechanisms
+		winset(src, "", "browser-options=byondstorage")
```
Comment (AffectedArc07) [inline code/modules/client/client_procs.dm:262]: How long are we gonna support old clients for? Not sure if `TODO` is applicable here.

File: tgui/packages/common/storage.js [hunk 2]
```diff
@@ -25,69 +24,44 @@ const testGeneric = (testFn) => () => {
...
-// See: https://superuser.com/questions/1080011
-// prettier-ignore
-const testLocalStorage = testGeneric(() => (
-  window.localStorage && window.localStorage.getItem
-));
+const testHubStorage = testGeneric(
+  () => window.hubStorage && window.hubStorage.getItem
+);
 
+// TODO: Remove with 516
```
Comment (AffectedArc07) [inline tgui/packages/common/storage.js:31]: Ditto

File: code/modules/client/client_procs.dm (hunk 1 above)
Comment (S34NW) [reply code/modules/client/client_procs.dm:262]: I'd prefer we drop them ASAP so TGUI doesn't have one foot in the grave with IE11 support

Comment (AffectedArc07) [reply code/modules/client/client_procs.dm:262]: Valid

**Comment by S34NW** [comment]
!merge_upstream

**Review by lewcc — APPROVED** [review]
🗣️ TEST IT ON LIVE

**Pull request merged**